### PR TITLE
17 recreate components

### PR DESCRIPTION
**Pull request by skif-cezar**
1. Added a header component.
2. Removed the list component.
3. Reused the button component.
4. Changed the design style of the project.

### CODE REVIEW

File: client/src/app/component/button/Button.tsx [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
 
 /**
  * Button component
  */
-const Button: React.FC = () => {
+const Button: React.FC<IButton> = (props: IButton) => {
   const btnClassName = clsx(styles.btn);
 
   return (
     <div>
```
Comment (tritonJS826) [inline client/src/app/component/button/Button.tsx]: unneccessary extra nesting

File: client/src/app/component/button/Button.tsx [hunk 2]
```diff
@@ -1,20 +1,26 @@
 import React from "react";
 import clsx from "clsx";
-import styles from "src/app/component/Button.module.scss";
+import styles from "src/app/component/button/Button.module.scss";
+
+interface IButton {
```
Comment (tritonJS826) [inline client/src/app/component/button/Button.tsx]: Wrong naming

File: client/src/app/component/header/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import {NavLink} from "react-router-dom";
+import clsx from "clsx";
+import styles from "src/app/component/header/Header.module.scss";
+import btnStyles from "src/app/component/button/Button.module.scss";
+
+const MAIN_PATH = "/";
+
+/**
+ * Header with links
```
Comment (tritonJS826) [inline client/src/app/component/header/Header.tsx]: "With links" -- this info must be incapsilated. Useless in comment

File: client/src/app/component/header/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ */
+const Header: React.FC = () => {
+  const headerClassName = clsx(styles.header);
+  const btnClassName = clsx(btnStyles.btn);
+
+  return (
+    <header className={headerClassName}>
+      <NavLink className={btnClassName} to={MAIN_PATH}>Main</NavLink>
+    </header>
```
Comment (tritonJS826) [inline client/src/app/component/header/Header.tsx]: "NavLink" with className "button" -- something strange) rename please one of them

File: client/src/app/component/questions/Questions.tsx [hunk 5]
```diff
@@ -17,7 +18,7 @@ const Questions: React.FC = () => {
   if (question) {
     if (questionIndex < question.length) {
       return (
-        <div>
+        <>
           <label><h3>{question[questionIndex]?.question}</h3></label>
```
Comment (tritonJS826) [inline client/src/app/component/questions/Questions.tsx]: place each tag on separate line

Comment (tritonJS826) [reply client/src/app/component/questions/Questions.tsx:22]: is question really optional

File: client/src/app/component/questions/Questions.tsx [hunk 6]
```diff
@@ -1,5 +1,6 @@
 import React from "react";
-import styles from "src/app/component/button/Button.module.scss";
+import Button from "src/app/component/button/Button";
+import Paragraph from "src/app/component/paragraph/Paragraph";
 import useHooks from "./UseHooks";
 
 /**
```
Comment (tritonJS826) [inline client/src/app/component/questions/Questions.tsx:6]: Lost formatting

File: client/src/app/component/questions/Questions.tsx [hunk 7]
```diff
@@ -1,5 +1,6 @@
 import React from "react";
-import styles from "src/app/component/button/Button.module.scss";
+import Button from "src/app/component/button/Button";
+import Paragraph from "src/app/component/paragraph/Paragraph";
 import useHooks from "./UseHooks";
```
Comment (tritonJS826) [inline client/src/app/component/questions/Questions.tsx]: Wrong naming

File: client/src/app/logic/mainPage/MainPage.tsx [hunk 8]
```diff
@@ -10,12 +10,12 @@ import {QUIZ_PATH, SETTINGS_PATH} from "src/app/App";
...
 const MainPage: React.FC = () => {
   const mainClassName = clsx(styles.main);
-  const btnStartClassName = clsx(btnStyles.btnStart, btnStyles.btn);
+  const btnClassName = clsx(btnStyles.btn);
 
   return (
     <main className={mainClassName}>
-      <NavLink className={btnStartClassName} to={QUIZ_PATH}>Start</NavLink>
-      <NavLink className={btnStartClassName} to={SETTINGS_PATH}>Settings</NavLink>
+      <NavLink className={btnClassName} to={QUIZ_PATH}>Start</NavLink>
```
Comment (tritonJS826) [inline client/src/app/logic/mainPage/MainPage.tsx]: "Navlink" with className "btn"

File: client/src/app/component/button/Button.tsx (hunk 1 above)
Comment (skif-cezar) [reply client/src/app/component/button/Button.tsx]: corrected

File: client/src/app/component/button/Button.tsx (hunk 2 above)
Comment (skif-cezar) [reply client/src/app/component/button/Button.tsx]: corrected

File: client/src/app/component/header/Header.tsx (hunk 3 above)
Comment (skif-cezar) [reply client/src/app/component/header/Header.tsx]: corrected

File: client/src/app/component/header/Header.tsx (hunk 4 above)
Comment (skif-cezar) [reply client/src/app/component/header/Header.tsx]: corrected

File: client/src/app/component/questions/Questions.tsx (hunk 5 above)
Comment (skif-cezar) [reply client/src/app/component/questions/Questions.tsx]: corrected

File: client/src/app/component/questions/Questions.tsx (hunk 6 above)
Comment (skif-cezar) [reply client/src/app/component/questions/Questions.tsx:6]: corrected

File: client/src/app/logic/mainPage/MainPage.tsx (hunk 8 above)
Comment (skif-cezar) [reply client/src/app/logic/mainPage/MainPage.tsx]: corrected

File: client/src/app/component/questions/Questions.tsx (hunk 7 above)
Comment (skif-cezar) [reply client/src/app/component/questions/Questions.tsx]: corrected

**Pull request merged**